crypto/evp/pmeth_lib.c

```diff
@@ -503,8 +503,12 @@ EVP_PKEY_CTX *EVP_PKEY_CTX_dup(const EVP_PKEY_CTX *pctx)
         if (pctx->op.kex.algctx != NULL) {
             if (!ossl_assert(pctx->op.kex.exchange != NULL))
                 goto err;
-            rctx->op.kex.algctx
-                = pctx->op.kex.exchange->dupctx(pctx->op.kex.algctx);
+
+            rctx->op.kex.algctx = NULL;
```

Comment:
Assigning to NULL is redundant due to the zalloc()

Reply:
I added that for readability. Should I remove it?

Reply:
I dont care too much, it is ok with it left in..

Reply:
I would actually prefer these NULL assignments to be removed. We don't do this anywhere else after zalloc, and its kind of the point of using zalloc in the first place.

Reply:
@mattcaswell @t8m removed those assignments.